hazelcast-sql/src/test/java/com/hazelcast/sql/impl/calcite/validate/types/InferTypesTestSupport.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.validate.types;
+
+import com.hazelcast.sql.impl.calcite.validate.HazelcastSqlConformance;
+import com.hazelcast.sql.impl.calcite.validate.HazelcastSqlValidator;
+import com.hazelcast.sql.impl.calcite.validate.operators.HazelcastSqlTrimFunction;
+import org.apache.calcite.config.CalciteConnectionConfig;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.sql.SqlBasicCall;
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.validate.SqlMoniker;
+import org.apache.calcite.sql.validate.SqlNameMatcher;
+import org.apache.calcite.sql.validate.SqlNameMatchers;
+import org.apache.calcite.sql.validate.SqlValidatorCatalogReader;
+import org.apache.calcite.sql.validate.SqlValidatorTable;
+
+import java.util.List;
+
+public class InferTypesTestSupport {
+    protected static RelDataType type(SqlTypeName typeName) {
+        return HazelcastTypeFactory.INSTANCE.createSqlType(typeName);
+    }
+
+    protected static RelDataType typeUnknown() {
+        return HazelcastTypeFactory.INSTANCE.createUnknownType();
+    }
+
+    protected static SqlCallBinding createBinding() {
+        SqlParserPos parserPos = new SqlParserPos(0, 0);
+        SqlCall call = new SqlBasicCall(new HazelcastSqlTrimFunction(), new SqlNode[0], parserPos);
+
+        HazelcastSqlValidator validator = new HazelcastSqlValidator(
+            new CatalogReader(),
+            HazelcastTypeFactory.INSTANCE,
+            HazelcastSqlConformance.INSTANCE
+        );
+
+        return new SqlCallBinding(validator, null, call);
+    }
+
+    private static class CatalogReader implements SqlValidatorCatalogReader {
```

Comment:
What about naming this `MockCatalogReader`?

Reply:
Renamed